src/main/scala/se/nimsa/dicom/streams/ParseFlow.scala

```diff
@@ -269,8 +269,10 @@ class ParseFlow(chunkSize: Int = 8192, stopTag: Option[Int] = None, inflate: Boo
           Some(SequencePart(tag, valueLength, state.bigEndian, state.explicitVR, bytes))
         else if (valueLength == indeterminateLength)
           Some(FragmentsPart(tag, valueLength, updatedVr2, state.bigEndian, state.explicitVR, bytes))
-        else
-          Some(HeaderPart(tag, updatedVr2, valueLength, isFmi = false, state.bigEndian, state.explicitVR, bytes))
+        else {
+          val realVr = if (state.explicitVR) vr else updatedVr2
```

Comment:
This is the bug fix. If explicit VR, use the original VR from the binary data, not the VR based on the guessing that takes place above with updatedVr1 and updatedVr2.